### DISCUSSION TITLE
Instructions on integration with FreeRTOS Cellular Interface

### DISCUSSION THREAD

**Discussion by NicoPowers** [Azure-Samples/iot-middleware-freertos-samples#178] [Q&A] [answered]
Hello,

I am working as an embedded engineer for a startup company and we are planning on using the Azure IoT Hub for our main platform, and I recently got the FreeRTOS Cellular Interface to work with our cellular modem, and as far as I know I should just have to define the transport layer to use the Cellular socket send/recv/open/close functions, but I am confused if I still need to use mbedtls or any other socket or TCP/IP library with this approach, as my modem is handling the socket connection and not my host processor.

Does anyone have any experience trying to integrate this functionality with a cellular modem? 

Thank you,

Nicolas

**Comment by wduraes** [comment, marked as answer]
Hi Nicolas, normally the Modems (cellular modems) take care of TLS and sockets, so you won't have to worry about them once you have defined the transport. If you managed to get this device connected to Azure and was able to send telemetry or any other Azure functionality, then you're good to go.
